core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -629,14 +645,14 @@ private[spark] class ExecutorAllocationManager(
      * Note: This is not thread-safe without the caller owning the `allocationManager` lock.
      */
     def pendingTasks(): Int = {
-      stageIdToNumTasks.map { case (stageId, numTasks) =>
-        numTasks - stageIdToTaskIndices.get(stageId).map(_.size).getOrElse(0)
+      stageAttemptToNumTasks.map { case (stageAttempt, numTasks) =>
+        numTasks - stageAttemptToTaskIndices.get(stageAttempt).map(_.size).getOrElse(0)
```

Comment:
I think we should not take tasks from previous stage attempts, which are already zombie,  into account, right ?

Reply:
In ```onStageCompleted``` processing, ```stageAttemptToNumTasks```, ```stageAttemptToTaskIndices``` removes the zombie stage, so pendingTasks does not calculate the zombie stage's task.

Reply:
I see.